lib/Email/MIME/Encode.pm

```diff
@@ -16,6 +17,10 @@ sub maybe_mime_encode_header {
     $header = lc $header;
 
     my $header_length = length($header) + length(": ");
+
+    return $val->as_mime_string($charset, $header_length)
```

Comment:
A look suggests that nothing should be passing anything to `maybe_mime_encode_header` other than the value `UTF-8`.  If this is the case, I think we should eliminate the argument.  Header objects that know how to encode their strings should be free to pick the appropriate encoding (which imho should either be ASCII or UTF-8 and nothing else), and shouldn't have to deal with someone asking for emoji to be encoded as KOI-8.

Any reason to keep this argument?